demo-terraform/aws/simple-ec2-with-s3-access/s3.tf

```diff
@@ -1,28 +1,51 @@
 
 resource "random_string" "bucket_suffix" {
-    upper = false
-    special = false
-    length = 32
+  upper   = false
+  special = false
+  length  = 32
 }
 
 resource "aws_s3_bucket" "bucket" {
```

Comment:
**Accurics remediation**
 
**[HIGH]: Ensure versioning is enabled for AWS S3 Buckets - Terraform Version 1.x**
 
**Vulnerability**
Enabling S3 versioning will enable easy recovery from both unintended user actions, like deletes and overwrites.
 
**Remediation**
In AWS Console - 
 1. Log into the AWS Management Console..
 2. Select the 'Services' option and search for S3.
 3. Scroll down the left navigation panel and choose 'Buckets'.
 4. Select the 'Bucket' that needs to be verified and click on its identifier(name) from the 'Bucket name' column.
 5. Click on the 'Properties' tab on the top menu.
 6. Check the 'Versioning' option under 'Properties' and if it's set to 'Suspend versioning' then S3 bucket versioning is not enabled for the selected S3 bucket.
 7. Repeat steps number 2 - 6 to verify other S3 buckets in the region.
 8. Select the 'S3 bucket' on which versioning needs to be enabled and click on the 'Properties' tab.
 9. Click on the 'Enable versioning'  option under 'Versioning'.
 10. Click 'Save'.
 In Terraform - 
 Set the 'versioning.enabled' attribute to 'true'.

source: accurics-